aodncore/pipeline/steps/harvest.py

```diff
@@ -72,83 +213,57 @@ def run(self, pipeline_files):
 
         for file_slice in deletion_slices:
             deletion_map = self.match_harvester_to_files(file_slice)
-            self.validate_harvester_mapping(file_slice, deletion_map)
+            validate_harvester_mapping(file_slice, deletion_map)
             self.run_deletions(deletion_map, self.tmp_base_dir)
 
         for file_slice in addition_slices:
             addition_map = self.match_harvester_to_files(file_slice)
-            self.validate_harvester_mapping(file_slice, addition_map)
+            validate_harvester_mapping(file_slice, addition_map)
             self.run_additions(addition_map, self.tmp_base_dir)
 
-    def match_harvester_to_files(self, file_list):
-        harvester_map = OrderedDict()
+    def match_harvester_to_files(self, pipeline_files):
+        validate_pipelinefilecollection(pipeline_files)
 
-        for harvester, config_item in self._config.trigger_config.items():
-            matched_files = PipelineFileCollection()
+        harvester_map = HarvesterMap()
 
+        for harvester, config_item in self._config.trigger_config.items():
             for event in config_item['events']:
-                for config_type, patterns in event.items():
+                extra_params = None
+                matched_files = PipelineFileCollection()
+
+                for config_type, value in event.items():
```

Comment:
Couldn't we replace this loop with something like this?
```python
    regex_list = event.get('regex', [])
    extra_params = event.get('extra_params')
    for pattern in regex_list:
        matched_files_for_pattern = ...
        ...

    if matched_files:
        event_obj = TriggerEvent(matched_files, extra_params)
```
(Note: I'm pretty sure `event.get()` returns `None` if the given key doesn't exist.)

Reply:
It's very likely we could replace it, but I didn't really look at that bit because it was in the "just works" category rather than "needs to change to support events" category.

I agree 100% though, the nested loop was basically pulled over directly from the Ruby, so should be refactored into a more Pythonic/less nested style. I'd probably prefer to do it in a follow up PR though, to avoid this one blowing out the scope? What do you reckon?

Reply:
:+1: Sounds good